src/Tooltip.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { forwardRef, memo, ReactElement } from "react";
+import type { Equals } from "tsafe";
+import { assert } from "tsafe/assert";
+import { symToStr } from "tsafe/symToStr";
+import { useAnalyticsId } from "./tools/useAnalyticsId";
+
+export type TooltipProps = TooltipProps.WithClickAction | TooltipProps.WithHoverAction;
+
+export namespace TooltipProps {
+    export type Common = {
+        description: string;
+        id?: string;
+        className?: string;
+    };
+
+    export type WithClickAction = Common & {
+        kind: "click";
+        children?: ReactElement | string;
+    };
+
+    export type WithHoverAction = Common & {
+        kind?: "hover";
+        children: ReactElement | string;
+    };
+}
+
+/** @see <https://components.react-dsfr.codegouv.studio/?path=/docs/components-tooltip> */
+export const Tooltip = memo(
+    forwardRef<HTMLSpanElement, TooltipProps>((props, ref) => {
+        const { id: id_prop, className, description, kind, children, ...rest } = props;
+        assert<Equals<keyof typeof rest, never>>();
+
+        const id = useAnalyticsId({
+            "defaultIdPrefix": "fr-tooltip",
+            "explicitlyProvidedId": id_prop
+        });
+
+        const displayChildren = (
+            children: ReactElement | string | undefined,
+            id: string
+        ): ReactElement => {
+            if (children === undefined) return <></>;
+            return typeof children === "string" ? (
+                <span aria-describedby={id} id={`tooltip-owner-${id}`}>
+                    {children}
+                </span>
+            ) : (
+                children &&
+                    React.cloneElement(children, {
+                        "aria-describedby": id,
+                        "id": `tooltip-owner-${id}`
+                    })
+            );
+        };
+
+        return (
+            <>
+                {props.kind === "click" ? (
+                    <span ref={ref}>
+                        {children === undefined ? (
+                            <button
+                                className="fr-btn--tooltip fr-btn"
+                                aria-describedby={id}
+                                id={`tooltip-owner-${id}`}
+                            >
+                                Information contextuelle
```

Comment:
should we add transation here ?

Reply:
should i fix the PR ?

Reply:
Yes you are welcome to!  
This is not my work, I reopenned the PR because I merged it too rearly

Reply:
You can see my code review here: https://github.com/codegouvfr/react-dsfr/pull/176#issuecomment-1720470241

Reply:
adressed most of the review comments

Reply:
For some reason i cannot make click behaviour work with custom children